api/openapi.yaml

```diff
@@ -0,0 +1,393 @@
+        '500':
+          description: Internal server error
+    delete:
+      tags:
+      - Users
+      responses:
+        '204':
+          description: User successfully deleted
+        '404':
+          description: User with this id not found
+        '500':
+          description: Internal server error
+  /users/{id}/followers:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Users
+      responses:
+        '200':
+          description: Follower users successfully retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '500':
+          description: Internal server error
+  /users/{id}/following:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Users
+      responses:
+        '200':
+          description: Following users successfully retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '500':
+          description: Internal server error
+  /messages:
+    get:
+      tags:
+      - Messages
+      parameters:
+      - name: poster
+        in: query
+        schema:
+          type: integer
+        required: false
+        description: id of user who posted messages
+      responses:
+        '200':
+          description: Successfully retrieved messages
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Message'
+        '500':
+          description: Internal server error
+    post:
+      tags:
+      - Messages
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Message'
+      responses:
+        '201':
+          description: Successfully created message
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Message'
+        '400':
+          description: Invalid Message object
+        '500':
+          description: Internal server error
+  /messages/{id}:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Messages
+      responses:
+        '200':
+          description: Successfully retrieved message
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Message'
+        '404':
+          description: Message with this id not found
+        '500':
+          description: Internal server error
+    put:
```

Comment:
We could probably take this out to disallow editing of messages

Reply:
Sweet, sounds good